modules/dnn/src/layers/padding_layer.cpp

```diff
@@ -221,7 +239,7 @@ class PaddingLayerImpl CV_FINAL : public PaddingLayer
         op->update_input_desc_paddings(*(op_const_paddings->getTensorDesc()));
         // set inputs : constant_values
         std::vector<int> constant_values_shape{1};
-        Mat constant_values_mat(1, 1, CV_32F, Scalar(paddingValue));
+        Mat constant_values_mat(1, 1, CV_32F, Scalar(paddingValueFloat));
```

Comment:
Is that really affect accuracy? `constant_values_mat` has `CV_32F` and should cast Scalar appropriately

Reply:
paddingValueFloat veriable is required only for Ngraph, because it requires a pointer to float value on initialization and probably doesn't copy the value. Here we can use paddingValue or (float)paddingValue, the result should be the same

Reply:
It copies so you can make it local.